---
title: "Class 14: Genome Informatics 1"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Asthma SNPs in MXL population

Read file from Ensemble

```{r}
mxl <- read.csv("373531-SampleGenotypes-Homo_sapiens_Variation_Sample_rs8067378.csv")
head(mxl)
```

How many of each genotype is in this set?
```{r}
table(mxl$Genotype..forward.strand.)
```

How many G|G genotypes are in this set?

```{r}
sum(mxl$Genotype..forward.strand. == "G|G")
```

% each genotype
```{r}
table(mxl$Genotype..forward.strand.) / nrow(mxl) * 100
```

## RNA-Seq

Fastq quality score

Q1. Does the first sequence have good quality?

```{r}
#first installed seqinr and gtools packages with install.packages("") in console
q <- "DDDDCDEDCDDDDBBDDDCC@"
library(seqinr)
library(gtools)
asc( s2c(q) ) - 33 #33 is the offset for fastqsanger
```

## Population Scale Analysis


Does the genotype for rs9067378 influence expression of the important asthma-linked genes we found?

```{r}
geno <- read.table("rs8067378_ENSG00000172057.6.txt")
head(geno)
```

How many of each genotype?
```{r}
table(geno$geno)
```

```{r}
inds.gg <- geno$geno == "G/G"
summary( geno$exp[inds.gg] ) #expression for G/G genotype
```

```{r}
inds.aa <- geno$geno == "A/A"
summary( geno$exp[inds.aa] ) #expression for A/A genotype
```

```{r}
inds.ag <- geno$geno == "A/G"
summary( geno$exp[inds.ag] ) #expression for A/A genotype
```

Some nice figure that shows the distribution of these three expression values:

```{r}
?boxplot
boxplot(exp ~ geno, geno)
```

```{r}
#library(ggplot2)
```

